tests/vmi_headless_test.go

```diff
@@ -20,230 +20,42 @@
-			})
-
-			It("[test_id:707]should create headless vmi without any issue", func() {
-				libvmops.RunVMIAndExpectLaunch(vmi, 30)
-			})
-
-			It("[test_id:714][posneg:positive]should not have vnc graphic device in xml", func() {
-				vmi = libvmops.RunVMIAndExpectLaunch(vmi, 30)
-
-				runningVMISpec, err := tests.GetRunningVMIDomainSpec(vmi)
-				Expect(err).ToNot(HaveOccurred(), "should get vmi spec without problem")
-
-				Expect(runningVMISpec.Devices.Graphics).To(BeEmpty(), "should not have any graphics devices present")
-			})
-
-			It("[test_id:737][posneg:positive]should match memory with overcommit enabled", func() {
-				vmi.Spec.Domain.Resources = v1.ResourceRequirements{
-					Requests: kubev1.ResourceList{
-						kubev1.ResourceMemory: resource.MustParse("100M"),
-					},
-					OvercommitGuestOverhead: true,
-				}
-				vmi = libvmops.RunVMIAndExpectLaunch(vmi, 30)
-
-				readyPod, err := libpod.GetPodByVirtualMachineInstance(vmi, testsuite.GetTestNamespace(vmi))
-				Expect(err).ToNot(HaveOccurred())
-				computeContainer := libpod.LookupComputeContainer(readyPod)
-
-				Expect(computeContainer.Resources.Requests.Memory().String()).To(Equal("100M"))
-			})
-
-			It("[test_id:2444][posneg:negative]should not match memory with overcommit disabled", func() {
-				vmi.Spec.Domain.Resources = v1.ResourceRequirements{
-					Requests: kubev1.ResourceList{
-						kubev1.ResourceMemory: resource.MustParse("100M"),
-					},
-				}
-				vmi = libvmops.RunVMIAndExpectLaunch(vmi, 30)
-
-				readyPod, err := libpod.GetPodByVirtualMachineInstance(vmi, testsuite.GetTestNamespace(vmi))
-				Expect(err).ToNot(HaveOccurred())
-				computeContainer := libpod.LookupComputeContainer(readyPod)
-
-				Expect(computeContainer.Resources.Requests.Memory().String()).ToNot(Equal("100M"))
-			})
-
-			It("[test_id:713]should have more memory on pod when headless", func() {
-				normalVmi := libvmifact.NewAlpine()
-
-				vmi = libvmops.RunVMIAndExpectLaunch(vmi, 30)
-				normalVmi = libvmops.RunVMIAndExpectLaunch(normalVmi, 30)
-
-				readyPod, err := libpod.GetPodByVirtualMachineInstance(vmi, testsuite.GetTestNamespace(vmi))
-				Expect(err).ToNot(HaveOccurred())
-				computeContainer := libpod.LookupComputeContainer(readyPod)
-
-				normalReadyPod, err := libpod.GetPodByVirtualMachineInstance(normalVmi, testsuite.GetTestNamespace(vmi))
-				Expect(err).ToNot(HaveOccurred())
-				normalComputeContainer := libpod.LookupComputeContainer(normalReadyPod)
+	autoattachGraphicsDevice := func(enable bool) libvmi.Option {
+		return func(vmi *v1.VirtualMachineInstance) {
+			vmi.Spec.Domain.Devices.AutoattachGraphicsDevice = &enable
+		}
+	}
 
-				memDiff := normalComputeContainer.Resources.Requests.Memory()
-				memDiff.Sub(*computeContainer.Resources.Requests.Memory())
+	It("[test_id:738][posneg:negative]should not connect to VNC but still be able to connect to console", decorators.Conformance, func() {
```

Comment:
sorry for not thinking of this in my first review, but I don't think this merits inclusion in `decorators.Conformance`. We already unittest that no `vnc` device is passed to libvirt. I'm not sure we must test libvirt not to invent a graphic device, or test that Alpine boots without a graphic device. But I am certain it does not belong in our slim, do-or-die, conformance suite.

I believe that every test that is added to `Conformance` should have a strong reason described in the commit message. The current explanation
```
The test showcase expected behavior for
user therefore let's cover it in Conformance.
```
can be applied to every proper e2e test.

Reply:
Hmm, I must confess I haven't think about this in this way. I was more focused on the fact that we would cover the expected behavior when user defines VMI with
```
...
autoattachGraphicsDevice : false  
...
```
so at least I should change the reasoning... 

>I believe that every test that is added to Conformance should have a strong reason described in the commit message.

Agree, I think generally the Conformance test should cover the expected behavior of the API (this test would therefore fit), thinks I don't see as such are:
1. Every permutation
2. Edge cases
3. Negative tests or more what the API doesn't mean, in other words we should define the API in terms of what it does and not what it doesn't do... 

Would you keep it with the reason of covering the expected behavior of the API field?

Reply:
> Would you keep it with the reason of covering the expected behavior of the API field?

I really don't like the fact that our API has defaults and the flags to disable the defaults, and defaults for the flags. It's two complex for humans to follow.

IMO `autoattachGraphicsDevice=false` is an edge case, that is properly tested by unit tests. The code here tests the the API does not create a device. I see no urgent need to pay time, electricity and labor to test it in every time we run Compliance.

Reply:
>I really don't like the fact that our API has defaults and the flags to disable the defaults, and defaults for the flags. It's two complex for humans to follow.

I see this option more like A or B, not about defaults, flags... How I interpret it is that in the past we did not expose the graphic options in order to explicitly deny the VDI (support by Kubevirt) and all the options you can do... I would personally run `autoattachGraphicsDevice=false` in my center when I just want to use it to run some web services... 

I think the use case is there and that is why I want the one test to be included.

Reply:
> I think the use case is there and that is why I want the one test to be included.

I don't think anybody doubts that there is a use case for headless VMs. I just state that it is a costly test with very little value, which IMO does not merit inclusion in `Conformance`.

Reply:
I agree with @dankenigsberg here, for a couple of reasons.

1. This specific test does not need to be an e2e test, I'd actually remove it and have a unit test that exercises that rest API endpoint.
2. Even if this test would check that libvirt does the correct thing (which it doesn't ) we should be removing it anyway as we don't have to test libvirt API, we should take that for granted and covered by libvirt own tests.
3. I see this as a niche setting which shouldn't be anyway part of a set of minimal conformance tests.

Reply:
@dankenigsberg @acardace  The PR is now only cleanup of headless test file. PTAL